### PR TITLE
ETFE 2470: Add trader name / ERN and 'change trader type' link

### PR DESCRIPTION
**Pull request by machinesleet**

### CODE REVIEW

File: app/controllers/actions/DataRetrievalAction.scala [hunk 1]
```diff
@@ -18,24 +18,29 @@ package controllers.actions
 
 import models.requests.{MovementRequest, OptionalDataRequest}
 import play.api.mvc.ActionTransformer
-import services.UserAnswersService
+import services.{ GetTraderKnownFactsService, UserAnswersService}
```
Comment (LewisMitchell) [inline app/controllers/actions/DataRetrievalAction.scala:21]:
```suggestion
import services.{GetTraderKnownFactsService, UserAnswersService}
```

File: app/views/IndexPage.scala.html [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 @()(implicit request: Request[_], messages: Messages)
 
-@layout(pageTitle = titleNoForm("emcs-tfe-change-destination-frontend"), showBackLink = false) {
+@layout(pageTitle = titleNoForm("emcs-tfe-change-destination-frontend"), maybeShowActiveTrader = None, showBackLink = false) {
```
Comment (LewisMitchell) [inline app/views/IndexPage.scala.html:21]: The user is technically authenticated on this page (according to the controller) so it should be possible to have the banner showing?

File: conf/application.conf [hunk 3]
```diff
@@ -47,38 +47,43 @@ play.http.errorHandler = "handlers.ErrorHandler"
...
-          protocol = http
-          host = localhost
-          port = 9250
+            protocol = http
+            host = localhost
+            port = 9250
         }
-  }
+    }
 }
```
Comment (LewisMitchell) [inline conf/application.conf:87]: Could you create (or link me to) the app-config-* PR's for these entries, please?

File: conf/application.conf [hunk 4]
```diff
@@ -101,6 +106,7 @@ features {
   welsh-translation = false
   allowListEnabled = false
   returnToLegacy = true
+  stub-get-trader-known-facts = true
```
Comment (LewisMitchell) [inline conf/application.conf:109]: Could you create (or link me to) the app-config-* PR's for these entry, please?

File: test/connectors/emcsTfe/referenceData/GetTraderKnownFactsHttpParserSpec.scala
Comment (LewisMitchell) [inline test/connectors/emcsTfe/referenceData/GetTraderKnownFactsHttpParserSpec.scala:1]: Could you add a test for when the JSON is invalid, please?

File: test/controllers/action/AuthActionSpec.scala [hunk 5]
```diff
@@ -30,16 +31,18 @@ import uk.gov.hmrc.auth.core.AffinityGroup.{Agent, Organisation}
...
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration.Duration
 
 class AuthActionSpec extends SpecBase with BaseFixtures with BeforeAndAfterAll {
 
   type AuthRetrieval = ~[~[~[Option[AffinityGroup], Enrolments], Option[String]], Option[Credentials]]
 
-  implicit val fakeRequest = FakeRequest()
+  implicit val fakeRequest = FakeRequest().withSession(SessionKeys.sessionId -> UUID.randomUUID().toString)
```
Comment (LewisMitchell) [inline test/controllers/action/AuthActionSpec.scala:45]: Why is this needed?

File: app/connectors/referenceData/GetTraderKnownFactsConnector.scala
Comment (LewisMitchell) [inline app/connectors/referenceData/GetTraderKnownFactsConnector.scala:1]: Could you add a unit test for this, please?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/connectors/referenceData/GetTraderKnownFactsConnector.scala
Comment (machinesleet) [reply app/connectors/referenceData/GetTraderKnownFactsConnector.scala:1]: There's an integration test for this in: it/connectors

File: app/views/IndexPage.scala.html (hunk 2 above)
Comment (machinesleet) [reply app/views/IndexPage.scala.html:21]: OK will have a look 👀

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Just one comment to resolve:
https://github.com/hmrc/emcs-tfe-change-destination-frontend/pull/8/files#r1403500190
